### DISCUSSION TITLE
Sponsorship Opportunity

### DISCUSSION THREAD

**Discussion by bkanuka** [fsspec/gdrivefs#34] [Show and tell]
Hi all,

I'm open to sponsoring someone to work on a couple issues within this project, mainly around authentication, documentation, and ease-of-use.  This can (and should!) be done in the open with proper PRs, tests, and code reviews. 

- Change the package name to not conflict with existing `gdrivefs` in pypi
- Add more options for Authentication including custom OAuth client ID and secret, and existing credentials (i.e. `google.auth.default()`)
- Add documentation for authentication methods

If interested, please contact me at my professional email:  bennett@climateengine.com 

cc: @martindurant

**Comment by efiop** [comment]
@bkanuka Have you considered https://github.com/iterative/pydrive2 ?

**Comment by bkanuka** [comment]
@efiop Well I completely missed that pydrive2 integrates with fsspec  :face_exhaling: 
https://docs.iterative.ai/PyDrive2/fsspec/

Thanks... although I'm a little embarrassed! haha

**Reply by efiop** [reply]
Not trying to sabotage gdrivefs development, just wanted to share in case pydrive2 has everything you need 🙂

**Comment by bkanuka** [comment]
I looked into PyDrive2 and it's close - there might be things that could be learned from the code, but this offer is absolutely still open.

The main issue is that PyDrive2 uses an old authentication method that ~will be~ is deprecated.  Being able to pass a project ID and [Credentials](https://google-auth.readthedocs.io/en/stable/reference/google.oauth2.credentials.html#google.oauth2.credentials.Credentials) to gdrivefs (or get them automatically from ADC) is still vital to our use case.

The other 2 issues I mentioned (naming and docs) are just to make this easier to use ;)

**Comment by martindurant** [comment]
We would be totally OK with renaming this repo and/or the package for the sake of being able to get it onto PyPI and conda. In fact - to be checked - it may just be the *project* name that needs to be changed. Likewise, basic documentation is only a small matter of effort, I would be supportive of any infra stuff it might need.

As far as auth mechanisms are concerned, this has historically proven tricky both with this project and gcsfs where the latter has more options with cloud service accounts and VM instance metadata. Google's documentation is less than clear, and their requirements on having a permanent oauth landing page in order to have an active client key: onerous. I would be very happy for *someone* to fix these things!
